src/data/nat/prime.lean

```diff
@@ -408,6 +408,27 @@ begin
       rw e, exact pow_dvd_pow _ l } }
 end
 
+/-- If `p` is prime, then `p^k` divides `p^l` implies that `k` is less than or equal to `l`. -/
+lemma prime_pow_dvd_prime_pow {p k l : ℕ} (pp : prime p) : p^k ∣ p^l ↔ k ≤ l :=
```

Comment:
This is true for `1 < p` right?

Reply:
Indeed, `pp` should be weakened. I'm surprised I can't find this in mathlib already.

Reply:
It's `nat.pow_le_iff_le_right`.

Reply:
(I only found this after I'd generalised it, worked out the right home for the new lemma, and discovered the old lemma a few lines below... Because it relies on `pow_right_strict_mono` the lemma isn't in the obvious place.)

Reply:
Okay, I've generalised and refactored this a bit. It's now in `data.nat.basic` as two lemmas:
```
lemma pow_dvd_pow_iff_pow_le_pow {x k l : ℕ} (w : 0 < x) : x^k ∣ x^l ↔ x^k ≤ x^l := ...

/-- If `1 < x`, then `x^k` divides `x^l` if and only if `k` is at most `l`. -/
lemma pow_dvd_pow_iff_le_right {x k l : ℕ} (w : 1 < x) : x^k ∣ x^l ↔ k ≤ l := ...
```

Reply:
I've found
```lean
src/data/nat/basic.lean:lemma pow_right_strict_mono {x : ℕ} (k : 2 ≤ x) : strict_mono (nat.pow x) :=
src/data/nat/basic.lean:lemma pow_left_strict_mono {m : ℕ} (k : 1 ≤ m) : strict_mono (λ (x : ℕ), x^m) :=
```